src/framework/auth/services/token/token.ts

```diff
@@ -121,3 +125,93 @@ export class NbAuthJWTToken extends NbAuthSimpleToken {
     return super.isValid() && (!this.getTokenExpDate() || new Date() < this.getTokenExpDate());
   }
 }
+
+const validateOAuth2Token = (data) => {
+  let result;
+  if (typeof data === 'string') {
+    try {
+      result = JSON.parse(data)
+    } catch (e) {}
+  }
+  return result;
+};
+
+/**
+ * Wrapper for OAuth2 token
+ */
+export class NbAuthOAuth2Token extends NbAuthSimpleToken {
+
+  static NAME = 'nb:auth:oauth2:token';
+
+  constructor(protected data: { [key: string]: string|number }|string = {}) {
+    // we may get it as string when retrieving from a storage
+    super(validateOAuth2Token(data));
+  }
+
+  /**
+   * Returns the token value
+   * @returns string
+   */
+  getValue(): string {
+    return this.token.access_token;
+  }
+
+  /**
+   * Returns the refresh token
+   * @returns string
+   */
+  getRefresh(): string {
+    return this.token.refresh_token;
+  }
+
+  /**
+   * Returns token payload
+   * @returns any
+   */
+  getPayload(): any {
+    if (!this.token || !Object.keys(this.token).length) {
+      throw new Error('Cannot extract payload from an empty token.');
+    }
+
+    return this.token;
+  }
+
+  /**
+   * Returns the token type
+   * @returns string
+   */
+  getType(): string {
+    return this.token.token_type;
+  }
+
+  /**
+   * Is data expired
+   * @returns {boolean}
+   */
+  isValid(): boolean {
+    return super.isValid() && (!this.getTokenExpDate() || new Date() < this.getTokenExpDate());
+  }
+
+  /**
+   * Returns expiration date
+   * @returns Date
+   */
+  getTokenExpDate(): Date {
+    if (!this.token.hasOwnProperty('expires_in')) {
+      return null;
+    }
+
+    const date = new Date();
+    date.setUTCSeconds(new Date().getUTCSeconds() + Number(this.token.expires_in));
+
+    return date;
+  }
+
+  /**
+   * Validate value and convert to string, if value is not valid return empty string
```

Comment:
I think you've missed token validation 😄 
Frankly speaking, I don't think it's a good idea to validate token and return empty string if it's invalid.
Lets keet method as it is and just fix the comment.